src/libraries/System.Text.Json/src/Resources/Strings.resx

```diff
@@ -396,8 +396,8 @@
   <data name="SerializationConverterWrite" xml:space="preserve">
     <value>The converter '{0}' wrote too much or not enough.</value>
   </data>
-  <data name="SerializerDictionaryKeyNull" xml:space="preserve">
-    <value>The dictionary key policy '{0}' cannot return a null key.</value>
+  <data name="NamingPolicyReturnNull" xml:space="preserve">
+    <value>The naming policy '{0}' cannot return null.</value>
```

Comment:
Should we include the naming policy in the message for the `PropertyNamingPolicy` used on CLR propertiesas well? Right now it says `The JSON property name for '{0}.{1}' cannot be null.`.

Reply:
Yes we could throw a more specific message (in a clean up PR), although right now it shouldn't be hard for a caller to tell what caused an exception with this message.